src/query_error.h

```diff
@@ -97,34 +98,43 @@ const char *QueryError_Strerror(QueryErrorCode code);
  *
  * Only has an effect if no error is already present
  */
-void QueryError_SetError(QueryError *status, QueryErrorCode code, const char *err);
+void QueryError_SetError(QueryError *status, QueryErrorCode code, const char *message);
 
 /** Set the error code of the query without setting an error string. */
 void QueryError_SetCode(QueryError *status, QueryErrorCode code);
 
 /** Set the error code using a custom-formatted string */
-void QueryError_SetErrorFmt(QueryError *status, QueryErrorCode code, const char *fmt, ...);
+void QueryError_SetErrorFmt(QueryError *status, QueryErrorCode code, const char* message, const char *fmt, ...);
+
+/**
+ * Set the error code using a custom-formatted string
+ * Only use this function if you are certain that no user data is leaked in the format string
+ */
+void QueryError_SetDataAgnosticErrorFmt(QueryError *status, QueryErrorCode code, const char *fmt, ...);
 
 /** Convenience macro to set an error of a 'bad argument' with the name of the argument */
 #define QERR_MKBADARGS_FMT(status, fmt, ...) \
   QueryError_SetErrorFmt(status, QUERY_EPARSEARGS, fmt, ##__VA_ARGS__)
```

Comment:
This needs to be updated according to the modification of `QueryError_SetErrorFmt`